NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs

```diff
@@ -177,29 +177,37 @@ private IEnumerable<Entity> SpawnEntitiesUsingRandomDistribution(EntitySpawnPoin
 
         if (chosenPrefab.Count == 0)
         {
-            yield break;
+            return [];
         }
 
+        List<Entity> spawnedEntities = [];
         if (worldEntityFactory.TryFind(chosenPrefab.ClassId, out UweWorldEntity uweWorldEntity))
         {
             for (int i = 0; i < chosenPrefab.Count; i++)
             {
                 // Random position in sphere is only possible after first spawn, see EntitySlot.Spawn
-                IEnumerable<Entity> entities = CreateEntityWithChildren(entitySpawnPoint,
+                List<Entity> entities = [.. CreateEntityWithChildren(entitySpawnPoint,
                                                                         chosenPrefab.ClassId,
                                                                         uweWorldEntity.TechType,
                                                                         uweWorldEntity.PrefabZUp,
                                                                         uweWorldEntity.CellLevel,
                                                                         uweWorldEntity.LocalScale,
                                                                         deterministicBatchGenerator,
                                                                         parentEntity,
-                                                                        i > 0);
-                foreach (Entity entity in entities)
+                                                                        i > 0)];
+
+                // Unparenting the new entity because it's directly under the cell root
+                entities[0].ParentId = null;
+                if (entities[0] is WorldEntity worldEntity)
                 {
-                    yield return entity;
+                    worldEntity.Transform.SetParent(null, true);
```

Comment:
Did the LocalPosition and LocalRotation get changed to be local to the CellRoot with this change? or at least swap Local transforms with global transforms?

Reply:
When first parented, the entity had their correct global position and rotation since the hierarchy is filled from the top, and thus, unparenting while locking the global position and rotation makes it so the entities are correctly placed